### PR TITLE
chore(developer): convert Server to ES Modules

### LINKED ISSUE #9672: chore(developer): convert developer/src/server to ESM

### PR DESCRIPTION
**Pull request by mcdurdin**
Fixes #9672.

Also rewrote build.sh as a builder script in order to simplify some of the build changes.

Makes /web a dependency of Keyman Developer Server, which is a significant improvement, but we need to consider if we want to use web's debug build as we did in the past.

# User Testing

TEST_SERVER: Open Keyman Developer, and try to debug a web keyboard. Ensure that Keyman Developer Server starts. Explore each function in Keyman Developer Server (web debugger), and ensure that it is working. Verify that the Keyman Developer Server tray icon is accessible and can be used to shut down Keyman Developer Server.

### CODE REVIEW

**Comment by bharanidharanj** [comment]
# Test Results

* **TEST_SERVER (FAILED):** Tested with the attached PR build (Keyman developer 17.0.185) in both Windows 10 (Native machine) and Windows 11 OS (VM) and here is my observation: 1. Opened an existing Khmer_Angkor keyboard project. 2. Clicked the Test Keyboard on Web button from the build view. A warning message 'Keyman Developer Server does not appear to be running. Try and start Keyman Developer Server now?' with Yes, No and Cancel buttons appear on the screen. 3. Clicking the Yes button, would appear 'Confirm' message. Clicking on the OK button did reappear the same Confirm message again. Noticed that the keyman developer server tray icon is missing on the taskbar.  Seems to be an issue. 

..Warning message 1
   

..Warning message 2 

   


..keyman developer tray icon is missing

File: developer/src/server/build.sh [hunk 1]
```diff
@@ -16,121 +10,68 @@ EX_USAGE=64
...
-# Install dependencies
-# ----------------------------------------
-
-# Check if Node.JS/npm is installed.
-type npm >/dev/null ||\
-    builder_die "Build environment setup error detected!  Please ensure Node.js is installed!"
-
-if (( install_dependencies )) ; then
+function configure_server() {
   verify_npm_setup
```
Comment (darcywong00) [inline developer/src/server/build.sh:52]: Does this handle installing dependencies? (install_dependences)

File: developer/src/server/package.json [hunk 2]
```diff
@@ -24,13 +22,13 @@
...
   "devDependencies": {
-    "@keymanapp/resources-gosh": "*",
     "@keymanapp/keyman-version": "*",
+    "@keymanapp/resources-gosh": "*",
     "@types/chai": "^4.3.0",
     "@types/express": "^4.17.13",
     "@types/mocha": "^9.1.0",
     "@types/multer": "^1.4.7",
-    "@types/node": "^17.0.0",
+    "@types/node": "^20.4.1",
```
Comment (darcywong00) [inline developer/src/server/package.json:31]: Are you on the current Node.js? (newer than 18.18 LTS)

File: developer/src/server/build.sh (hunk 1 above)
Comment (mcdurdin) [reply developer/src/server/build.sh:52]: Yes, that's now done in configure, and verify_npm_setup does the necessary `npm install`.

File: developer/src/server/package.json (hunk 2 above)
Comment (mcdurdin) [reply developer/src/server/package.json:31]: Running on 18.x; the types needed to be brought up to 18.x minimum and most types refs were 20.4.1 so I just went with that. It's only used for typescript engine within vscode, so it's not critical.

**Comment by mcdurdin** [comment]
I've addressed the problem that caused Keyman Developer Server to fail to start. Can you please retest?

@keymanapp-test-bot retest

**Comment by bharanidharanj** [comment]
> I've addressed the problem that caused Keyman Developer Server to fail to start. Can you please retest @bharanidharanj ?
> 
> @keymanapp-test-bot retest

@mcdurdin Okay, I will do it.

**Review by darcywong00 — APPROVED** [review]
lgtm

**Comment by mcdurdin** [comment]
Thanks Bharani. Could you run the same test on current alpha because I
think that issue is pre-existing. If so, can you open it as a separate
issue and mark this as passed?

**Comment by mcdurdin** [comment]
I have moved the 'Show Console' problem into #9694.

TEST_SERVER (PASSED)

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.187-alpha](https://keyman.com/downloads/releases/alpha/17.0.187)

**Pull request merged**